pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -211,37 +253,41 @@ func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1a
 	// TODO Update the MasterUserRecord with NSTemplateTier values
 	// SEE https://jira.coreos.com/browse/CRT-74
 
+	compliantUsername, err := r.generateCompliantUsername(userSignup)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusFailedToCreateMUR, err,
+			"Error generating compliant username for %s", userSignup.Spec.Username)
+	}
+
+	labels := map[string]string{toolchainv1alpha1.MasterUserRecordUserIDLabelKey: userSignup.Name}
+
 	mur := &toolchainv1alpha1.MasterUserRecord{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      userSignup.Spec.CompliantUsername,
+			Name:      compliantUsername,
 			Namespace: userSignup.Namespace,
+			Labels:    labels,
 		},
 		Spec: toolchainv1alpha1.MasterUserRecordSpec{
 			UserID:       userSignup.Name,
```

Comment:
We should stop using UserID field, shouldn't we? UserID is already set in the label. This filed is a duplicate.